app.js

```diff
@@ -0,0 +1,468 @@
+    }
+
+    file = filePathFromIRI(physicalFileIRI);
+
+    if (!existsSync(file)) {
+      throw new Error('File not found on disk: ' + JSON.stringify(file));
+    }
+    const clamscanResult = await clamscanFile(file);
+    const fileHasVirus = clamscanResult.isInfected;
+    switch (fileHasVirus) {
+      case false:
+        ret.stixMalwareAnalysis.result = 'benign';
+        break;
+      case true:
+        ret.stixMalwareAnalysis.result = 'malicious';
+        ret.stixMalwareAnalysis.resultName = JSON.stringify(
+          clamscanResult.viruses,
+        );
+        break;
+      case null:
+        throw new Error('clamscan JS returned null: Unable to scan');
+        break; // eslint-disable-line no-unreachable
+      default:
+        throw new Error('Unexpected return value from clamscan JS');
+    }
+  } catch (e) {
+    ret.error = e;
+  }
+  ret.stixMalwareAnalysis.analysisEnded = new Date();
+  ret.lookups = {
+    requestedFileIRI: fileIRI,
+    physicalFileIRI,
+    physicalFilePath: file,
+  };
+  console.log(ret);
+  return ret;
+}
+
+/**
+ * Calls the clamscan JS library to scan a file for viruses.
+ *
+ * @async
+ * @function
+ * @param {String} path - Path of file to scan.
+ * @returns {Object} As per clamscan 2.1.2:
+ * - `file` (string) The original `filePath` passed into the `isInfected`
+ *                   method.
+ * - `isInfected` (boolean) **True**: File is infected;
+ *                          **False**: File is clean.
+ *                          **NULL**: Unable to scan.
+ * - `viruses` (array) An array of any viruses found in the scanned file.
+ */
+async function clamscanFile(path) {
+  console.log('Running virus scan on file: ' + JSON.stringify(path));
+  const scanner = await new NodeClam().init({
+    clamscan: {
+      // Do not use clamscan binary because it loads database on every run.
+      active: false,
+    },
+    clamdscan: {
+      socket: '/var/run/clamav/clamd.ctl', // Unix domain socket
+      host: false, // Do not connect via TCP interface
+      port: false, // Do not connect via TCP interface
+      localFallback: false, // Do not use local preferred binary to scan if socket/tcp fails
+      active: true,
+    },
+    preference: 'clamdscan',
+  });
+  const result = await scanner.isInfected(path);
+  console.log(result);
+  return result;
+  // For now, error handling will be the responsibility of the function caller.
+}
+
+/**
+ * Gets the physical file IRI associated to a virtual/logical file IRI
+ */
+async function getPhysicalFileIRI(logicalFileIRI) {
+  const result = await query(`
+    PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
+    SELECT ?physicalFile
+    WHERE {
+      GRAPH ?g {
+        ?physicalFile nie:dataSource ${sparqlEscapeUri(logicalFileIRI)} .
+      }
+    }
+  `);
+  if (result.results.bindings.length)
+    // `[0]` is based on the assumption that, even if there are triples
```

Comment:
using `distinct` won't need this comment. I can also image a lot of other egde cases (e.g. derived files) but let's not dwell upon that.